File: src/bindings/ruby/Slave.rb
require 'msg'
require 'RubyProcess'
require 'RubyTask'
include MSG

class Slave < RbProcess
  
  def initialize()
    super()
  end
  
  #for Testing
  def msg_main2(args)
    info("Hello From Slave")
  end
  
  
  
  
  
 # msg_main : that function that will be executed when Running Simulation
  def msg_main(args)
    info("Hello From Slave")
    s_mailbox = "slave>>" + args[0]

    while true
        
       info("Ready to Receive Task")
       task = RbTask.receive(s_mailbox)
       task_name = RbTask.name(task)
       info ("Task Received : " + task_name)
      if (task_name == "finalize")
	info("Slave" + s_mailbox + "got finalize msg")
	break
      end
      info("Slave " + s_mailbox + " ...Processing" + RbTask.name(task))
       RbTask.execute(task)
    end
    info("Slave " + s_mailbox +  "I'm Done , See You !!")
    end
    
  end

File: examples/lua/console/slave.lua


-- Slave Function ---------------------------------------------------------
function Slave(...)

  if #arg ~= 1 then
    error("Wrong number of arguments (got " .. #arg .. ", expected 1: slave_id)")
  end

  local my_mailbox = "slave " .. arg[1]
  simgrid.info("Hello from lua, I'm a poor slave with mbox: " .. my_mailbox)

  while true do

    local task = simgrid.task.recv(my_mailbox);
    --print(task)
    local task_name = task:get_name()
    if (task:get_name() == "finalize") then
      simgrid.info("Slave '" .. my_mailbox .. "' got finalize msg");
      break
    end
    --local tk_name = simgrid.task.get_name(tk) 
    simgrid.info("Slave '" .. my_mailbox .. "' processing " .. task:get_name())
    simgrid.task.execute(task)
    simgrid.info("Slave '"  .. my_mailbox .. "': task " .. task:get_name() .. " done")
  end -- while

  simgrid.info("Slave '" .. my_mailbox .. "': I'm Done . See You !!");

end 
-- end_of_slave
